### ISSUE TITLE
Overwrite when calling createSourceFile

### ISSUE THREAD

**Issue by lazarljubenovic** [dsherret/ts-simple-ast#273]
After [a quick search](https://github.com/dsherret/ts-simple-ast/search?q=overwrite&type=Code&utf8=%E2%9C%93), I've found that apparently this option was considered for some operations but not for all, such as `save` and `saveSync`. It would be nice if we could overwrite when saving a file.

Seems to me like I have to manually delete the file, maybe there's another way?

**Comment by dsherret** [comment]
Can you post a short snippet of code that shows the problem? It should be overwriting.

**Comment by lazarljubenovic** [comment]
Oh? It tells me:

```
/home/lazar/ts-ast/node_modules/ts-simple-ast/dist/factories/CompilerFactory.js:154
        throw new errors.InvalidOperationError(prefixMessage + "A source file already exists at the provided file path: " + filePath);
        ^

Error: A source file already exists at the provided file path: /home/lazar/ts-ast/demo2-counter/compiled/RootPageComponent.factory.js
```

I'm just creating a file, writing a function into it and then saving it. I'll create a full repro tonight.

**Comment by dsherret** [comment]
Oh ok. That makes sense. That error happens on createSourceFile (not on save).

I’ll think about this... probably will add an options object as the third parameter.

**Comment by lazarljubenovic** [comment]
Oh, I didn't even notice where it happens. Sorry for the confusion and thanks! :sweat_smile:

**Comment by dsherret** [comment]
This will be in the next release (tomorrow evening probably).

**Comment by dsherret** [comment]
This is available now in 10.0.1.

**Comment by FaBeyyy** [comment]
> 
> 
> This is available now in 10.0.1.

Doesnt work for me sadly... if overwrite is set to true it does not throw an error but also doesnt overwrite the file.
